### ISSUE TITLE
Use hex-encoded Public Key Hash instead of Base64 to avoid / char in SPIFFE-ID

### ISSUE THREAD

**Issue by caleblloyd** [bloomberg/spire-tpm-plugin#7]
[SPIFFE-ID paths](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#22-path) may be hierarchial, and using the base64 encoded Public Key Hash results in SVIDs that look like:

`spiffe://domain.test/spire/agent/tpm/c1XbxrikL+sgdCw7hKFlmh2/y/QenHiIfLazBlsG/yQ=`

This appears as if the path is hierarchial, since base64 includes `/` in it's character set

Additionally, #4 proposes storing authorized Public Key Hashses as an empty file on the filesystem.  Since `/` is a directory separator on Linux, it cannot be represented in a file

Hex encoding for SHA256 representations is fairly standard practice, and would alleviate these issues

**Comment by pawalt** [comment]
Yup I think this is a good move. This will be a small change, so let's wait until I can get the tests PR gets merged to do this so we don't have to deal with too many merge conflicts.

**Comment by caleblloyd** [comment]
Great! I have a follow-up PR ready once testing is merged
